From d94f8b171dc2d59e63835f7fabd090effdbd9d45 Mon Sep 17 00:00:00 2001
From: Igor Grinberg <igrinberg@habana.ai>
Date: Tue, 12 Jan 2021 21:41:19 +0200
Subject: [PATCH] [SW-20719] preboot: clear stack area

Add stack area to be cleared upon Preboot load.
This makes a fare assumption the stack section follows the BSS section.

Change-Id: I2651ce3103af50048ab53d2e275230cd6ff56115
Signed-off-by: Igor Grinberg <igrinberg@habana.ai>
---

diff --git a/preboot/crt0_preboot.S b/preboot/crt0_preboot.S
index f2608b1..b329174 100644
--- a/preboot/crt0_preboot.S
+++ b/preboot/crt0_preboot.S
@@ -20,9 +20,9 @@
 	mov	sp, __stack_top
 	mov	fp, sp
 
-/* Clear BSS */
+	/* Clear BSS and stack areas */
 	mov	r5, __bss_start
-	sub	r6, __bss_end, r5
+	sub	r6, __stack_top, r5
 	lsr.f	lp_count, r6, 2
 	lpnz	1f
 	st.ab   0, [r5, 4]
@@ -34,9 +34,9 @@
 	bic	sp, x0, #0xf	/* 16-byte alignment for ABI compliance */
 	mov	x29, sp		/* Set FP (x29) */
 
-/* Clear BSS */
+	/* Clear BSS and stack areas */
 	ldr	x0, =__bss_start
-	ldr	x1, =__bss_end
+	ldr	x1, =__stack_top
 clear_loop:
 	str	xzr, [x0], #8
 	cmp	x0, x1
